Gems/LmbrCentral/Code/Source/Shape/BoxShape.cpp

```diff
@@ -84,11 +84,14 @@ namespace LmbrCentral
 
     void BoxShape::OnTransformChanged(const AZ::Transform& /*local*/, const AZ::Transform& world)
     {
-        m_currentTransform = world;
-        m_intersectionDataCache.InvalidateCache(InvalidateShapeCacheReason::TransformChange);
-        ShapeComponentNotificationsBus::Event(
-            m_entityId, &ShapeComponentNotificationsBus::Events::OnShapeChanged,
-            ShapeComponentNotifications::ShapeChangeReasons::TransformChanged);
+        if (m_currentTransform != world)
```

Comment:
Is this comparison really going to be false?

The notification is called `OnTransformChanged`, which should be trusted that it actually calls when the transform has changed and it's not the same one.

Reply:
My experience with OnChanged events is that you should _not_ trust that it's only called when a value actually changes. I've seen multiple systems that will trigger an On*Changed simply because a value was set at _all_, not that it was set to something _different_.

Reply:
Yes, these events get sent very often "just in case", causing a cascade of other events costing us a lot in performance. 
We actually used to have a similar optimization in RigidBodyComponent but it uses a different system now.

If everyone is happy about the change, I'll make the same in the other shape classes.

Reply:
That's exactly what we used to have in rigid body component, but I thought to be on the safe side + minimize the overhead of this check. 
IsClose() should work too